common.blocks/button/button.bemhtml

```diff
@@ -1,4 +1,11 @@
 block('button')(
+    def()(function() {
+        var isRealButton = (['button', 'input'].indexOf(apply('tag')) >= 0) &&
```

Comment:
1) What if we move the content of `tag` mode inside `def`?

~~~js
def()(() => {
  this.ctx.tag || (this.ctx.tag = 'button');
  // ···
})
~~~

2) I think we can omit `input` case check and leave only `tag === 'button'`. You can't use `input` as a tag of the `button`, as it must have `button__text` and `button__icon` in it's content and this is unachievable.